apps/condo/domains/contact/access/ContactRole.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Generated by `createschema contact.ContactRole 'name:Text'`
+ */
+
+const { throwAuthenticationError } = require('@condo/domains/common/utils/apolloErrorFormatter')
+const {
+    queryOrganizationEmployeeFor,
+    checkPermissionInUserOrganizationOrRelatedOrganization,
+} = require('@condo/domains/organization/utils/accessSchema')
+const get = require('lodash/get')
+const { getById } = require('@core/keystone/schema')
+const { queryOrganizationEmployeeFromRelatedOrganizationFor } = require('../../organization/utils/accessSchema')
+
+async function canReadContactRoles ({ authentication: { item: user } }) {
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+
+    if (user.isAdmin || user.isSupport) return {}
+
+    return {
+        OR: [
+            { organization_is_null: true },
+            { organization: queryOrganizationEmployeeFor(user.id) },
+            { organization: queryOrganizationEmployeeFromRelatedOrganizationFor(user.id) },
+        ],
+    }
+}
+
+async function canManageContactRoles ({ authentication: { item: user }, originalInput, operation, itemId }) {
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+    if (user.isAdmin || user.isSupport) return true
+
+    let organizationId
+    if (operation === 'create') {
+        organizationId = get(originalInput, ['organization', 'connect', 'id'])
+    } else if (operation === 'update') {
+        if (!itemId) return false
+        organizationId = get(originalInput, ['organization', 'connect', 'id'])
+
+        if (!organizationId) {
+            const contactRole = await getById('ContactRole', itemId)
+            organizationId = get(contactRole, 'organization')
+        }
+    }
+
+    return checkPermissionInUserOrganizationOrRelatedOrganization(user.id, organizationId, 'canManageContacts')
```

Comment:
probably we can create a new `canManageContactRoles` permission

Reply:
It's looks different access level to manage contacts and manage their roles

Reply:
done